e2e/cypress/integration/channel/more_channels_spec.js

```diff
@@ -141,6 +141,42 @@ describe('Channels', () => {
         // * Assert that archived channel doesn't show up in LHS list
         cy.get('#publicChannelList').should('not.contain', testChannel.display_name);
     });
+
+    it('MM-T1702 Search works when changing public/archived options in the dropdown', () => {
+        cy.apiAdminLogin();
+        cy.apiUpdateConfig({
+            TeamSettings: {
+                ExperimentalViewArchivedChannels: true,
+            },
+        });
+
+        // # Go to LHS and click "More..." under Public Channels group
+        cy.get('#publicChannelList').should('be.visible').within(() => {
+            cy.findByText('More...').scrollIntoView().should('be.visible').click();
+        });
+
+        // * Dropdown should be visible, defaulting to "Public Channels"
+        cy.get('#channelsMoreDropdown').should('be.visible').click().within((el) => {
+            cy.findByText('Public Channels').should('be.visible').click();
+            cy.wrap(el).should('contain', 'Show: Public Channels');
+        });
+
+        // * Users should be able to type and search
+        cy.get('#searchChannelsTextbox').should('be.visible').type('a').wait(TIMEOUTS.HALF_SEC);
+        cy.get('#moreChannelsList').should('be.visible');
+        cy.get('#searchChannelsTextbox').clear();
+
+        cy.get('#moreChannelsModal').should('be.visible').within(() => {
+            // * Users should be able to switch to "Archived Channels" list
+            cy.get('#channelsMoreDropdown').should('be.visible').and('contain', 'Show: Public Channels').click().within((el) => {
+                // # Click on archived channels item
+                cy.findByText('Archived Channels').should('be.visible').click();
+
+                // * Modal should show the archived channels list
+                cy.wrap(el).should('contain', 'Show: Archived Channels');
```

Comment:
Please add verification for "On step #2, matching Public and Private archived channels are shown."